pytorch_lightning/trainer/supporters.py

```diff
@@ -217,6 +217,13 @@ def __next__(self) -> Any:
     def __len__(self) -> Union[int, float]:
         return self.length
 
+    @staticmethod
+    def get_len(cycle_iterator: "CycleIterator") -> Union[int, float]:
+        if has_len(cycle_iterator.loader):
+            return len(cycle_iterator.loader)
+
+        return float("inf")
```

Comment:
```suggestion
        return len(cycle_iterator.loader
```
why not just this?

And it is probably problematic to have both `__len__` and the new `get_len` defined. Can we combine them?

Reply:
Done !